hipchat.go

```diff
@@ -86,15 +86,22 @@ type ErrorResponse struct {
 type Client struct {
 	AuthToken string
 	BaseURL   string
+	ProxyURL  string
```

Comment:
Instead of having a specified ProxyURL, let’s do this:
```
type Client struct {
  AuthToken string
  BaseURL string
  Timeout  time.Duration
  Transport http.RoundTripper
}
```

Then instead of using http.Get, you create a client using these properties:

```
client := &http.Client{
  Transport: c.Transport,
  Timeout:   c.Timeout,
}
```

and make the request that way.